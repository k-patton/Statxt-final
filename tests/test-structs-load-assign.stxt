struct types {
	int i;
	char c;
	float f;
	string s;
};

struct types2 {
	int i;
	char c;
	float f;
	string s;
};

int main() {
	
	struct types t1;
	struct types2 t2;

	t1.i = 1;
	t2.i = t1.i;

	t1.c = 'a';
	t2.c = t1.c;

	t1.f = 3.14;
	t2.f = t1.f;

	t1.s = "hello world!";
	t2.s = t1.s;

	int i1;
	int i2;
	int i3;

	float f1;
	float f2;
	float f3;
	float f4;

	i1 = t2.i - t1.i;
	i2 = t2.i / 2;
	i3 = t2.i * t1.i;
	f1 = t1.f + t2.f;
	f2 = t1.f - t2.f - 5.1;
	f3 = t2.f / 2.4;
	f4 = t1.f * 5.2;

	print(t2.i);
	printchar(t2.c);
	printf(t2.f);
	printstr(t2.s);

	print(i1);
	print(i2);
	print(i3);
	printf(f1);
	printf(f2);
	printf(f3);
	printf(f4);

	return 0;
}
