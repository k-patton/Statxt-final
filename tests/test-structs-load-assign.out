1
a
3.14
hello world!
0
0
1
6.28
-5.1
1.30833
16.328
